nano/node/transport/tcp.cpp

```diff
@@ -370,13 +370,14 @@ void nano::transport::tcp_channels::ongoing_keepalive ()
 		channel->send (message);
 	}
 	// Attempt to start TCP connections to known UDP peers
+	nano::tcp_endpoint invalid_endpoint (boost::asio::ip::address_v6::any (), 0);
 	if (!node.network_params.network.is_test_network () && !node.flags.disable_udp)
 	{
 		size_t random_count (std::min (static_cast<size_t> (6), static_cast<size_t> (std::ceil (std::sqrt (node.network.udp_channels.size ())))));
 		for (auto i (0); i <= random_count; ++i)
 		{
 			auto tcp_endpoint (node.network.udp_channels.bootstrap_peer (nano::tcp_realtime_protocol_version_min));
-			if (find_channel (tcp_endpoint) == nullptr)
+			if (find_channel (tcp_endpoint) == nullptr && tcp_endpoint != invalid_endpoint)
```

Comment:
I guess invalid check should be before find () for better performance

Reply:
Yep